### PR TITLE
Update Makefile

### LINKED ISSUE #727: Update Makefile
### Description

Update the Makefile to support running/managing a Lisk Core node easily with the `make` command.

### Acceptance Criteria

- Fix the apk package version pinning in Dockerfile (Ref: https://github.com/LiskHQ/lisk-service/pull/1581)
- Add the following commands to the Makefile
  - `build`: Build the latest docker image from the source-code
  - `start`: Start a new Lisk Core container
  - `stop`: Stop a running Lisk Core container
  - `logs`: Print the container logs

### PR DESCRIPTION
**Pull request by vardan10**
### What was the problem?

This PR resolves #727

### How was it solved?

- [x] Updated Makefile
- [x] Updated Dockerfile

### How was it tested?

- [x] Local

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ ARG NODEJS_VERSION=16
 
 FROM node:$NODEJS_VERSION-alpine AS builder
 
-RUN apk add --no-cache alpine-sdk=~1.0-r1 python3=~3.10.10-r0 libtool=~2.4.7-r1 autoconf=~2.71-r1 automake=~1.16.5-r1 rust=~1.64.0-r2 cargo=~1.64.0-r2 cmake=~3.24.4-r0 clang15-libs=~15.0.7-r0 clang15-dev=~15.0.7-r0 clang15=~15.0.7-r0 rustfmt=~1.64.0-r2 linux-headers=~5.19.5-r0 && \
+RUN apk add --no-cache alpine-sdk=~1.0 python3=~3.10 libtool=~2.4.7-r1 autoconf=~2.71-r1 automake=~1.16.5-r1 rust=~1.64.0-r2 cargo=~1.64.0-r2 cmake=~3.24 clang15-libs=~15.0.7-r0 clang15-dev=~15.0.7-r0 clang15=~15.0.7-r0 rustfmt=~1.64.0-r2 linux-headers=~5.19.5-r0 && \
```
Comment (sameersubudhi) [inline docker/Dockerfile:7]:
```suggestion
RUN apk add --no-cache alpine-sdk=~1.0 python3=~3.10 libtool=~2.4 autoconf=~2.71 automake=~1.16 rust=~1.64 cargo=~1.64 cmake=~3.24 clang15-libs=~15.0 clang15-dev=~15.0 clang15=~15.0 rustfmt=~1.64 linux-headers=~5.19 && \
```

File: Makefile [hunk 2]
```diff
@@ -17,6 +17,18 @@ else ifneq ($(findstring canary,$(CORE_VERSION)),)
 CORE_CHANNEL := canary
 endif
 
+build:
+	cd docker && docker build -f Dockerfile --build-arg NODEJS_VERSION=$(shell cat ./.nvmrc ) -t=lisk/core:$(CORE_VERSION) ..
```
Comment (sameersubudhi) [inline Makefile:21]:
```suggestion
	docker build -f ./docker/Dockerfile --build-arg NODEJS_VERSION=$(shell cat ./.nvmrc ) -t=lisk/core:$(CORE_VERSION) .
```

File: Makefile [hunk 3]
```diff
@@ -46,3 +43,42 @@ release: dist/lisk-core-v$(CORE_VERSION)
...
+stop:
+	docker stop lisk-core
+
+logs:
+	docker logs lisk-core
+
+logs-live:
+	docker logs lisk-core --follow
+
+run: check-args
```
Comment (nagdahimanshu) [inline Makefile:76]: Do we still need `run` because we already have `start` defined here?

File: Makefile [hunk 4]
```diff
@@ -46,3 +43,42 @@ release: dist/lisk-core-v$(CORE_VERSION)
 dist/lisk-core-v$(CORE_VERSION):
 	npx oclif-dev pack --targets=linux-x64,darwin-x64
 endif
+
+build: build-image build-local
+
+build-image:
+	docker build -f ./docker/Dockerfile --build-arg NODEJS_VERSION=$(shell cat .nvmrc) --tag=lisk/core .
```
Comment (vardan10) [inline Makefile:50]: Should we add version while tagging image?
```suggestion
	docker build -f ./docker/Dockerfile --build-arg NODEJS_VERSION=$(shell cat .nvmrc) --tag=lisk/core:$(CORE_VERSION) .
```

File: Makefile (hunk 3 above)
Comment (sameersubudhi) [reply Makefile:76]: `run` is to let users treat it as a cli tool. It's not necessary but just wanted to offer that option as well.

File: Makefile (hunk 4 above)
Comment (sameersubudhi) [reply Makefile:50]: No, this is mostly for the development purpose. The exact version tagging is done and uploaded to the DockerHub by the DevOps post release.

**Pull request merged**